gitops/snapshot.go

```diff
@@ -125,6 +125,9 @@ var (
 // MarkSnapshotAsPassed updates the AppStudio Test succeeded condition for the Snapshot to passed.
 // If the patch command fails, an error will be returned.
 func MarkSnapshotAsPassed(adapterClient client.Client, ctx context.Context, snapshot *applicationapiv1alpha1.Snapshot, message string) (*applicationapiv1alpha1.Snapshot, error) {
+	if !IsSnapshotNotStarted(snapshot) && HaveAppStudioTestsSucceeded(snapshot) {
```

Comment:
Having this conditionals in setter is very suspicious, it may mean we have a serious issues with code flow.

If snapshot is not ready to be marked as passed, this function shouldn't be even called at first place

Reply:
How about now ?  The !IsSnapshotNotStarted(snapshot)? is not needed any more.

Reply:
Why is not `HaveAppStudioTestsSucceeded` before we even call MarkSnapshotPassed?

Reply:
I think that the intent here is to first check if the Snapshot has the status condition already set to what we want to set it before unnecessarily calling the `adapterClient.Status().Patch()` and updating the metrics. If it's already `Succeeded`, we don't want to waste time and API calls to try to set it again to `Succeeded`.

That said, looking at it now I agree it would be more idiomatic to check the state of the Snapshot's status before calling these functions.

Reply:
Adding these checks  is to ensure no repetition action during the requests are requeued  and the integration service is restarted.  When the snapshot is already being marked to some status,  we don't want to mark and count it to the same status again.